lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1210 @@
+    if answers:
+        return create_questionnaire_response_item(question_id, answers)
+
+    return None
+
+
+def create_follow_up_encounter(record: REDCapRecord,
+                               patient_reference: dict,
+                               site_reference: dict,
+                               initial_encounter_reference: dict) -> tuple:
+    """
+    Returns a FHIR Encounter resource entry and reference for a follow-up
+    encounter
+    """
+    if not record.get('fu_timestamp'):
+        return None, None
+
+    encounter_identifier = create_identifier(
+        system = f"{INTERNAL_SYSTEM}/encounter",
+        value = f"{REDCAP_URL}{record.project.id}/{record['record_id']}/{record['redcap_event_name']}/{record['redcap_repeat_instance']}_follow_up"
+    )
+    encounter_class_coding = create_coding(
+        system = "http://terminology.hl7.org/CodeSystem/v3-ActCode",
+        code = "HH"
+    )
+    encounter_reason_code = create_codeable_concept(
+        system = "http://snomed.info/sct",
+        code = "390906007",
+        display = "Follow-up encounter"
+    )
+
+    # YYYY-MM-DD HH:MM in REDCap
+    encounter_date = record['fu_timestamp'].split()[0]
+    encounter_resource = create_encounter_resource(
+        encounter_identifier = [encounter_identifier],
+        encounter_class = encounter_class_coding,
+        encounter_date = encounter_date,
+        patient_reference = patient_reference,
+        location_references = [site_reference],
+        reason_code = [encounter_reason_code],
+        part_of = initial_encounter_reference
+    )
+
+    return create_entry_and_reference(encounter_resource, "Encounter")
+
+
+def create_follow_up_questionnaire_response(record: dict, patient_reference: dict,
+                                            encounter_reference: dict) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for the follow-up
+    encounter.
+
+    Note: `fu_which_activites` is misspelled on
+    purpose to match the misspelling of the fields in the REDCap project.
+    """
+    boolean_questions = [
+        'fu_illness',
+        'fu_change',
+        'result_changes',
+        'fu_household_sick'
+    ]
+
+    integer_questions = [
+        'fu_days_illness',
+        'fu_number_sick'
+    ]
+
+    string_questions = [
+        'fu_fever',
+        'fu_headache',
+        'fu_cough',
+        'fu_chills',
+        'fu_sweat',
+        'fu_throat',
+        'fu_nausea',
+        'fu_nose',
+        'fu_tired',
+        'fu_ache',
+        'fu_breathe',
+        'fu_diarrhea',
+        'fu_rash',
+        'fu_ear',
+        'fu_eye',
+        'fu_smell_taste',
+        'fu_feel_normal',
+        'fu_care',
+        'fu_hospital_ed',
+        'fu_work_school',
+        'fu_activities',
+        'fu_which_activites',
+        'fu_test_result',
+        'fu_behaviors_no',
+        'fu_behaviors_inconclusive',
+        'fu_behaviors',
+        'fu_1_symptoms',
```

Comment:
This function (I can't highlight the entire thing in GitHub UI) is an exact duplicate of the one in SCAN and should be imported from a shared module.

Reply:
Yes. Future work is to refactor the two ETLs.